packages/golden-layout/src/items/Stack.ts

```diff
@@ -276,8 +266,9 @@ export default class Stack extends AbstractContentItem {
     const insertBefore =
       this._dropSegment === 'top' || this._dropSegment === 'left';
     const hasCorrectParent =
-      (isVertical && this.parent.isColumn) ||
-      (isHorizontal && this.parent.isRow);
+      this.parent instanceof RowOrColumn &&
```

Comment:
I am making a slight assumption adding the `instanceof RowOrColumn` check here, since `AbstractContentItem` has an optional `isColumn` and `isRow` prop. Technically it wouldn't have to be said instance to have passed this check in the past. The other piece is the call to `addChild` on line 200 which does assume the 3rd argument in `addChild` which is only present in `RowOrColumn`